org.adempiere.ui/src/org/compiere/apps/form/Archive.java

```diff
@@ -28,43 +34,66 @@
 public class Archive {
 	
 	/**	Window No			*/
-	public int         m_WindowNo = 0;
+	protected int         m_WindowNo = 0;
 	/**	The Archives		*/
-	public MArchive[]	m_archives = new MArchive[0];
+	protected MArchive[]	m_archives = new MArchive[0];
 	/** Archive Index		*/
-	public int			m_index = 0;
+	protected int			m_index = 0;
 	/** Table direct		*/
-	public int 		m_AD_Table_ID = 0;
+	protected int 		m_AD_Table_ID = 0;
 	/** Record direct		*/
-	public int 		m_Record_ID = 0;
+	protected int 		m_Record_ID = 0;
 	
 	/**	Logger			*/
-	public static final CLogger log = CLogger.getCLogger(Archive.class);
+	protected static final CLogger log = CLogger.getCLogger(Archive.class);
 
+	/** optional trx name **/
+	private String m_trxName = null;
+	
+	/**
+	 * 
+	 * @return KeyNamePair array of AD_Process records
+	 */
 	public KeyNamePair[] getProcessData()
 	{
 		// Processes
 		final MRole role = MRole.getDefault(); // metas
-//		int AD_Role_ID = Env.getAD_Role_ID(Env.getCtx());
 		
 		boolean trl = !Env.isBaseLanguage(Env.getCtx(), "AD_Process");
 		String lang = Env.getAD_Language(Env.getCtx());
-		// TODO: ASP - implement process and window access ASP control
 		String sql = "SELECT DISTINCT p.AD_Process_ID,"
 				+ (trl ? "trl.Name" : "p.Name ")
 			+ " FROM AD_Process p INNER JOIN AD_Process_Access pa ON (p.AD_Process_ID=pa.AD_Process_ID) "
 			+ (trl ? "LEFT JOIN AD_Process_Trl trl on (trl.AD_Process_ID=p.AD_Process_ID and trl.AD_Language=" + DB.TO_STRING(lang) + ")" : "") 
-			+ " WHERE "+role.getIncludedRolesWhereClause("pa.AD_Role_ID", null) // metas: use included roles

```

Comment:
I have not tested yet - but removing the role.getIncludedRolesWhereClause from the where is not changing the access for users with included roles?

Reply:
Thanks @hengsin - understood

Tests with Archive feature in the UI went fine.

What's the goal of replacing the role.getIncludedRolesWhereClause here?
The question is because I see the same code in getTableData, and that method is also used in cmd_query, so, wondering if is a performance improvement or the goal is something different

I'm approving this as all tests passed